compiler/symbol.go

```diff
@@ -292,10 +294,39 @@ func (c *compilerContext) parsePragmas(info *functionInfo, f *ssa.Function) {
 				if len(parts) != 3 {
 					continue
 				}
-				c.checkWasmImport(f, comment.Text)
+				if f.Blocks != nil {
+					// Defined functions cannot be exported.
+					c.addError(f.Pos(), "can only use //go:wasmimport on declarations")
+					continue
+				}
+				c.checkWasmImportExport(f, comment.Text)
 				info.exported = true
 				info.wasmModule = parts[1]
 				info.wasmName = parts[2]
+			case "//go:wasmexport":
+				if f.Blocks == nil {
+					c.addError(f.Pos(), "can only use //go:wasmexport on definitions")
+					continue
+				}
+				if len(parts) != 2 {
+					c.addError(f.Pos(), fmt.Sprintf("expected one parameter to //go:wasmimport, not %d", len(parts)-1))
+					continue
+				}
+				name := parts[1]
+				if name == "_start" || name == "_initialize" {
```

Comment:
So `//export` remains for exports like `_start` and `_initialize` that don’t get the goroutine / scheduler dance?

Reply:
It is, the check has been moved into `case "//go:wasmimport":` above.